boxcli/args.go

```diff
@@ -4,17 +4,37 @@
 package boxcli
 
 import (
+	"fmt"
 	"path/filepath"
 
+	"github.com/fatih/color"
 	"github.com/pkg/errors"
 )
 
 // Functions that help parse arguments
 
 // If args empty, defaults to the current directory
 // Otherwise grabs the path from the first argument
-func pathArg(args []string) string {
+func pathArg(args []string, flags *configFlags) string {
```

Comment:
### 1
If you want to keep the logic simpler, I think it'd be ok to error if both flag and path as an argument are specified at the same time:
+ path-as-arg: warning
+ flag: no warning
+ both: error
### 2
I think you somehow need to distinguish between the flag being explicitly set and not set. At least I was assuming that:
- `devbox add foo` looks for `devbox.json` in a parent directory if it's not in the current directory. 
- `devbox add foo -c /my/project` looks for a `devbox.json` in `/my/project` _only_ (and not in any of it's parent directories). By extension that means `devbox add foo -c "."` only looks in the current directory and not in the parents.
- `devbox add foo -c /my/project/my-file.json` treats `my-file.json` as the config file.

That said, open to a different approach. How were you thinking it should work?

Reply:
> I think it'd be ok to error if both flag and path as an argument are specified at the same time:

👍🏾 

> That said, open to a different approach. How were you thinking it should work?

I'd been assuming we'd match the current behavior i.e. `devbox add foo -c /my/project` will treat `/my/project` as the starting point of the search and traverse up to the parent directories.

Happy to change that logic to your proposal. It _does_  feel more clear to not do the magic search if the user is specifying the config-path directly. 

How do you feel about doing that as a followup PR? I fear a bigger refactoring is needed because the search-parent-dirs logic is inside `devbox.go`. If its simple, I'll fold the change into this PR.